metadata-ingestion/tests/unit/sql_queries/test_sql_queries.py

```diff
@@ -426,3 +439,246 @@ def test_create(self, entry_dict, entry_config, expected_query_entry, should_rai
+            for proc in processors
+            if isinstance(proc, partial) and proc.func == auto_incremental_lineage
+        )
+        assert auto_incremental_processor.args[0] is False
+
+    def test_workunit_processors_default_behavior(
+        self, pipeline_context, temp_query_file
+    ):
+        """Test workunit processors with default configuration (incremental_lineage=False)."""
+        config = SqlQueriesSourceConfig(
+            query_file=temp_query_file,
+            platform="snowflake",
+            # incremental_lineage not specified, should default to False
+        )
+
+        source = SqlQueriesSource(pipeline_context, config)
+        processors = source.get_workunit_processors()
+
+        assert len(processors) == 2
+
+        # Verify that auto_incremental_lineage is configured with False by default
+        auto_incremental_processor = next(
+            cast(partial, proc)
+            for proc in processors
+            if isinstance(proc, partial) and proc.func == auto_incremental_lineage
+        )
+        assert auto_incremental_processor.args[0] is False
+
+    def test_workunit_generation_structure(self, pipeline_context, temp_query_file):
+        """Test that MCPs are generated with proper structure."""
+        config = SqlQueriesSourceConfig(
+            query_file=temp_query_file, platform="snowflake", incremental_lineage=True
+        )
+
+        source = SqlQueriesSource(pipeline_context, config)
+
+        # Generate MCPs (these will be converted to workunits by the processors)
+        mcps = list(source.get_workunits_internal())
+
+        # Should generate some MCPs (exact number depends on SQL aggregator behavior)
+        assert len(mcps) >= 0  # At minimum, no errors should occur
+
+        # All items should be MCPs
+        for mcp in mcps:
+            # Should be MetadataChangeProposalWrapper objects
+            assert hasattr(mcp, "aspectName") or hasattr(mcp, "aspect")
+
+    @pytest.mark.parametrize("incremental_lineage", [True, False])
+    def test_config_inheritance(
+        self, pipeline_context, temp_query_file, incremental_lineage
+    ):
+        """Test that incremental_lineage config is properly inherited by the source."""
+        config = SqlQueriesSourceConfig(
+            query_file=temp_query_file,
+            platform="snowflake",
+            incremental_lineage=incremental_lineage,
+        )
+
+        source = SqlQueriesSource(pipeline_context, config)
+
+        # Verify config is properly set
+        assert source.config.incremental_lineage == incremental_lineage
+
+        # Verify processors reflect the config
+        processors = source.get_workunit_processors()
+        auto_incremental_processor = next(
+            cast(partial, proc)
+            for proc in processors
+            if isinstance(proc, partial) and proc.func == auto_incremental_lineage
+        )
+        assert auto_incremental_processor.args[0] == incremental_lineage
```

Comment:
is this duplicating the tests: `test_workunit_processors_with_incremental_lineage_enabled` and `test_workunit_processors_with_incremental_lineage_disabled`

if so, we may keep this one and remove the others

also, we may delete `test_workunit_processors_default_behavior` if we set

```
    @pytest.mark.parametrize("incremental_lineage", [None, True, False])
```